### DISCUSSION TITLE
Use text in an InfoBadge

### DISCUSSION THREAD

**Discussion by ItsTheSky** [amwx/FluentAvalonia#569] [Q&A] [answered]
Hi there. It seems the "Value" property of InfoBadge only supports `int`, is there a way I can provide a text (I would need to display a version in there, e.g. `v1.2.5`)?
Thanks!

**Comment by robloo** [comment]
> is there a way I can provide a text (I would need to display a version in there, e.g. v1.2.5)?

No there isn't. `InfoBadge` is a port of the WinUI control and behaves the same. It is designed to show a changeable count -- although I agree in some cases just text would be useful (asterisk, infinity, etc.).

You are free to create a new control using the `InfoBadge` code as a base that supports your use case. It shouldn't be too difficult.

**Reply by ItsTheSky** [reply]
Gotcha, I'm not good enough to create a new control based on InfoBadge as I'm not sure what to copy and chance. Any chance it cold come in the official FA version?

**Comment by timunie** [comment]
@amwx would you be okay with a PR changing it to `object`? This would also allow to add `DataTemplates` and thus symbol icons for example.

**Reply by robloo** [reply]
The only issue is styling. With a number we can control the size and appearance. With generic content there is a lot more responsibility on the app to make sure everything fits and looks correct. It's more difficult to use in that case.

**Reply by timunie** [reply]
well, if I bind an int, nothing changes. If I place any content, up to the dev to ensure apperance is okay.

**Reply by amwx** [reply, marked as answer]
Changing to `object`, no. That opens the control up to things its not designed for, plus you end up boxing a value type just to convert it to a string - of which the control is normally designed for. 
If anything, I'd add a `ValueString` property and then the displayed value would be whichever is set, with the original `Value` property taking priority. However, this is not something I'm looking at in the 2.1 time frame.

There is a workaround you could use for now. Use a `FontIconSource` and with the string you want to display, and set that as the IconSource to the InfoBadge. I haven't tested this, but it theoretically should work. There are also various display modes you can add a styling class for - see the styles file for InfoBadge, or the Sample app. Just keep in mind, this control is not meant for long strings, so it should be kept to a few characters max.

**Reply by ItsTheSky** [reply]
Yup thanks, got it working with a font icon, it looks clean! Is there any chance to have a sort of "universal badge" that'll take control instead? Or at least string? (those can really be useful a bit everywhere I think, just my opinion tho)
![image](https://github.com/amwx/FluentAvalonia/assets/50667548/64daeffd-f15c-4d5c-a397-fadd15412bed)

While I'm here, is there a way to change its style (I mean if there's a built-it style like a transparent one, or do we have to override it ourselves?)
